include/pybind11/common.h

```diff
@@ -260,13 +261,18 @@ struct overload_hash {
     }
 };
 
+typedef std::pair<std::vector<void *>, const std::vector<std::type_index> *> implicit_instances;
+
 /// Internal data struture used to track registered instances and types
 struct internals {
     std::unordered_map<std::type_index, void*> registered_types_cpp; // std::type_index -> type_info
     std::unordered_map<const void *, void*> registered_types_py;     // PyTypeObject* -> type_info
     std::unordered_map<const void *, void*> registered_instances;    // void * -> PyObject*
     std::unordered_set<std::pair<const PyObject *, const char *>, overload_hash> inactive_overload_cache;
     std::forward_list<void (*) (std::exception_ptr)> registered_exception_translators;
+    std::unordered_map<std::type_index, std::vector<std::pair<PyTypeObject *, void *(*)(void *)>>> implicit_conversions_cpp;
```

Comment:
It would be nicer to have single unordered_map here, whose value is a ``struct`` with both the implicit conversion constructor and the destructor.